src/main/java/pl/psnc/dei/service/TranscriptionPlatformService.java

```diff
@@ -208,8 +213,26 @@ public JsonObject fetchTranscriptionUpdate(Transcription transcription) {
 					}
 				})
 				.block();
+		return JSON.parse(retrieveCurrentTranscription(transcription, response));
+	}
 
-		return JSON.parse(response);
+	private String retrieveCurrentTranscription(Transcription transcription, String enrichments){
+		String transcriptionId = transcription.getTp_id();
+		JSONArray enrichmentsJson = null;
+		try {
+			JSONParser parser = new JSONParser();
+			enrichmentsJson = (JSONArray) parser.parse(enrichments);
+		} catch (ParseException e) {
+			e.printStackTrace();
+		}
+
+		for(Object en: enrichmentsJson){
+			JSONObject json = (JSONObject) en;
+			if(json.getAsString("AnnotationId").equals(transcriptionId)){
```

Comment:
I would add some null pointer check for this...what about situation when there is actually no AnnotationId in the json?